source/fleetprovisioning/FleetProvisioning.cpp

```diff
@@ -494,7 +494,8 @@ bool FleetProvisioning::ProvisionDevice(shared_ptr<SharedCrtResourceManager> fpC
         LOG_INFO(TAG, "Fleet Provisioning Feature has been started.");
 
         bool didSetup = FileUtils::CreateDirectoryWithPermissions(keyDir.c_str(), S_IRWXU) &&
-                        FileUtils::CreateDirectoryWithPermissions(Config::DEFAULT_CONFIG_DIR, S_IRWXU);
+                        FileUtils::CreateDirectoryWithPermissions(
+                            Config::DEFAULT_CONFIG_DIR, S_IRUSR | S_IWUSR | S_IXUSR | S_IRGRP | S_IROTH | S_IXOTH);
```

Comment:
Isn't this the same as `Config::CONFIG_DIR` eg (octal 745 / `rwxr--r-x`). If so, then maybe better to reference the constant.

Reply:
Since this change needs to be applied at multiple places, I will have a separate PR for updating this and adding unit tests for the same.